com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -904,8 +979,7 @@ protected virtual void Update()
 
                 if (!CanCommitToTransform)
                 {
-#if NGO_TRANSFORM_DEBUG
-                    if (m_CachedNetworkManager.LogLevel == LogLevel.Developer)
+                    if (m_CachedNetworkManager.LogLevel == LogLevel.Developer && m_AuthorityCheckDelay < Time.realtimeSinceStartup)
```

Comment:
We seem to be going to a lot of effort here with this delay all to hide the warning message.  What if we simply don't show it.  Is it something useful to users; something they can act on?  Can the test coverage be enough where we don't feel the need to emit it?

Reply:
If a user has a bug in their code where they are trying to make changes to the NetworkTransform when they don't have authority, then without this message they may never know that they are doing this or have no easy way to check to make sure they are not trying to change a transform without authority.

_It would be difficult to write a tests for the scenario where a user's code is potentially changing the position, rotation, or scale of a NetworkObject that does not have the authority to make the change.  This would be more of an "improper usage" scenario which is better to provide some tool/way to detect this._

Reply:
As a side note:
This PR really turned into more of a "fix several issues" in associated areas, but most importantly fixes the delta rotation not exceeding the minimum threshold.  Due to the way it was being done, we would be sending delta updates to rotation due to false-positive results because we were not finding the shortest Euler angle between local and state transform values but just the absolute value of the difference between the local and state transform values.

The additional stuff for the warnings only happens when LogLevel.Developer, so it shouldn't have any real impact on performance while also providing a way for users to determine if they are trying to change transform values without authority in their code base.

Reply:
I'm somewhat ok with telling users that they are writing to a transform that they shouldn't be writing to - if we can do it in a water-tight way.  I'm not thrilled about the 1-tick delay and associated code that goes with it.  Can we fundamentally explain why it's precisely 1 tick, or did we add a 1 tick delay and the problem went away?  If the latter, how do we know it's 1 and only 1 tick for all platforms / situations?  

I also not crazy about rubber-banding back the transform, but only in developer mode.

Reply:
I believe 1 tick is a gross overshoot of the required time.  Let me see if there is a better way to handle this.